### ISSUE TITLE
Minutes: GitHub Review Meetings [every Thursday]

### ISSUE THREAD

**Issue by moul** [gnolang/meetings#18]
Longterm issue to keep track of discussions and PRs related to the GitHub Review Meetings.

Instructions: https://github.com/gnolang/meetings#weekly-review-meeting

## Ideas

- [ ] write a script making lists of issues/PRs with useful metrics (amount of changes, amount of comments, age, stale score, amount of dependents, etc) to ease choosing the best topics.
- [ ] write a script making lists of issues/PRs to generate public reports.
- [ ] consider having templated/canned answers and more labels, to make the review process faster and more consistent (maybe a bad idea).

**Comment by moul** [comment]
# Session #1 -- Jul, 27th 2023

Experimental session, we managed to discuss ~5 PRs, worked on improving labels and pipeline, and discussed more in details about the concept of memPkg VS Dir VS fs package.

Updated PRs & issues:
- https://github.com/gnolang/gno/pull/955
- https://github.com/gnolang/gno/pull/978
- https://github.com/gnolang/gno/pull/953
- https://github.com/gnolang/gno/pull/987
- https://github.com/gnolang/gno/pull/945
- https://github.com/gnolang/gno/pull/874
- https://github.com/gnolang/gno/pull/960
- https://github.com/gnolang/gno/pull/896
- https://github.com/gnolang/gno/pull/981
- https://github.com/gnolang/gno/pull/942
- https://github.com/gnolang/gno/issues/986

**Comment by moul** [comment]
# Session 2 -- Aug, 3rd 2023

Updated PRs & issues: 
- https://github.com/gnolang/gno/pull/825#issuecomment-1664264452 -- need changes

**Comment by zivkovicmilos** [comment]
# Session 4 -- Aug, 17th 2023
---
Updated PRs & issues:
- https://github.com/gnolang/gno/pull/781
- https://github.com/gnolang/gno/pull/1052
- https://github.com/gnolang/gno/pull/1010
- https://github.com/gnolang/gno/pull/964
- https://github.com/gnolang/gno/pull/1048
- https://github.com/gnolang/gno/pull/960

**Comment by thehowl** [comment]
# Session 5 -- Aug, 24th 2023

Updated PRs & issues:

- https://github.com/gnolang/gno/pull/660 -- commented
- https://github.com/gnolang/gno/pull/832 -- close
- https://github.com/gnolang/gno/pull/992 -- needs further discussion (cc/ @piux2 to open a related RFC issue)

Also discussed: possibility to improve meeting by also doing a filtering/labeling/triaging part (on input of @piux2 @mvertes), including issues. This is also in light of conversation to better prioritise what we need in order to reach mainnet.

**Comment by moul** [comment]
# Session 6 -- Aug, 31st 2023

Updated PRs & issues:
- https://github.com/gnolang/gno/pull/1080 -- discussed, merged
- https://github.com/gnolang/gno/pull/1078
- https://github.com/gnolang/gno/pull/1077
- https://github.com/gnolang/gno/pull/1089 -- merged
- https://github.com/gnolang/gno/pull/546 -- postponed (depends on @jaekwon)
- https://github.com/gnolang/gno/pull/1076 -- additional comments

Also discussed: 
- Using a new kanban workflow for triage (issues & PRs) -> PoC here: https://github.com/orgs/gnolang/projects/8

**Comment by thehowl** [comment]
## Session 7 -- Sep, 7th 2023

Updated PRs & issues:

* https://github.com/gnolang/gno/pull/875#pullrequestreview-1615846012 -- reviewed & commented
* https://github.com/gnolang/gno/pull/1036 -- merged
* https://github.com/gnolang/gno/pull/957 -- closed, redirected to new PR
* https://github.com/gnolang/gno/pull/1048 -- commented, discussed

**Comment by moul** [comment]
## Session 8 -- Sep, 14th 2023

Updated PRs & issues:

* https://github.com/gnolang/gno/pull/781 -- commented
* https://github.com/gnolang/gno/pull/1112 -- marked as blocked
* https://github.com/gnolang/gno/pull/1111 -- updated, merged
* https://github.com/gnolang/gno/pull/1048 -- discussed, needs more investigation

**Comment by gfanton** [comment]
## Session 9 -- Oct, 5th 2023

Updated PRs & issues:

* https://github.com/gnolang/gno/pull/1179 -- commented
* https://github.com/gnolang/gno/pull/1177 -- marked as blocked (see comments)
* https://github.com/gnolang/gno/pull/1117 -- merged
* https://github.com/gnolang/gno/pull/1077 -- merged (waiting for CI)
* https://github.com/gnolang/gno/pull/1178 -- made some more reviews, waiting for additional ones
* https://github.com/gnolang/gno/pull/891 -- made some more reviews, waiting for additional ones
* https://github.com/gnolang/gno/pull/1158 -- merged

**Comment by thehowl** [comment]
## Session 10 -- Oct, 12th 2023

Updated PRs & issues:

- https://github.com/gnolang/gno/pull/1103 -- merged
- https://github.com/gnolang/gno/pull/1134 -- needs a couple reviews, but can likely be merged expeditely
- https://github.com/gnolang/gno/pull/1216 -- awaiting @piux2 review
- https://github.com/gnolang/gno/pull/1202 -- needs more reviews
- https://github.com/gnolang/gno/pull/1078 -- merged
- https://github.com/gnolang/gno/pull/1185 -- awaiting @thehowl @piux2 review
- https://github.com/gnolang/gno/pull/1035 -- awaiting @moul + one of @gnolang/tech-staff review
- https://github.com/gnolang/gno/pull/1199 -- merged
- https://github.com/gnolang/gno/pull/1213 -- awaiting @piux2 review
- https://github.com/gnolang/gno/pull/1221 -- awaiting @piux2 review
- https://github.com/gnolang/gno/pull/1226 -- awaiting @piux2 review
- https://github.com/gnolang/gno/pull/1047 -- needs thorough discussion

**Comment by piux2** [comment]
## Session 11 -- Nov, 2th 2023

Updated PRs & issues:

- https://github.com/gnolang/gno/pull/1046/ -- need  discussion with @moul @zivkovicmilos 
- https://github.com/gnolang/gno/pull/1244/ -- need  discussion with @moul 
- https://github.com/gnolang/gno/pull/1047 -- need discussion with @moul 
- https://github.com/gnolang/gno/pull/1246 -- @thehowl @deelawn @mvertes review
- https://github.com/gnolang/gno/pull/1196 -- @thehowl review 
- https://github.com/gnolang/gno/pull/1307 -- merged

**Comment by thehowl** [comment]
## Session 12 -- Nov, 9th 2023

Also discussed: strategy for [#1196](https://github.com/gnolang/gno/pull/1196), gno by example and playground.

Updated PRs & issues:

- https://github.com/gnolang/gno/pull/1046 -- pending @moul review; almost ready to merge!
- https://github.com/gnolang/gno/pull/1212 -- decided how to progress
- https://github.com/gnolang/gno/pull/1213 -- awaiting for @jefft0 to make a last test following @piux2 instructions, then ready to merge
- https://github.com/gnolang/gno/pull/1221 -- discussed, pending re-review from @piux2 
- https://github.com/gnolang/gno/pull/1270 -- merged
- https://github.com/gnolang/gno/pull/1313 -- merged
- https://github.com/gnolang/gno/pull/1328 -- merged
- https://github.com/gnolang/gno/pull/1311 -- merged

**Comment by thehowl** [comment]
## Session 13 -- Nov, 16th 2023

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1329#pullrequestreview-1735034270 -- on hold waiting addition of golden tests.
- https://github.com/gnolang/gno/pull/1378 -- merge
- https://github.com/gnolang/gno/pull/1213 -- updated
- https://github.com/gnolang/gno/pull/1297 -- merge
- https://github.com/gnolang/gno/pull/1349 -- merge
- https://github.com/gnolang/gno/pull/1330 -- needs a txtar test

**Comment by thehowl** [comment]
## Session 14 -- Nov, 23rd 2023

Updated PRs and issues:
- https://github.com/gnolang/gno/pull/1213 -- merge
- https://github.com/gnolang/gno/pull/1001 -- merge 
- https://github.com/gnolang/gno/pull/1235 -- updated, need review again
- https://github.com/gnolang/gno/pull/1138 -- merge
- https://github.com/gnolang/gno/pull/1153 -- merge
- https://github.com/gnolang/gno/pull/1233 -- merge
- https://github.com/gnolang/gno/pull/1329 -- merge

**Comment by thehowl** [comment]
## Session 15 -- Nov, 30th 2023

Updated PRs and issues:

- https://github.com/gnolang/awesome-gno/pull/41 -- deferred to devrels
- https://github.com/gnolang/gno/pull/1367 -- needs further discussion
- https://github.com/gnolang/gno/pull/1048 -- lengthy discussion and decision making!

**Comment by gfanton** [comment]
## Session 16 -- Dec, 7th 2023

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1350 - waiting for https://github.com/gnolang/gno/pull/1305
- https://github.com/gnolang/gno/pull/1359 - merged
- https://github.com/gnolang/gno/pull/1385 - merged
- https://github.com/gnolang/tx-exports/pull/7 - merged
- https://github.com/gnolang/game-of-realms/pull/14 - merged

**Comment by piux2** [comment]
## Session 17 -- Dec, 14th 2023

Updated PRs and issues:

- https://github.com/gnolang/awesome-gno/pull/41 - merged
- https://github.com/gnolang/gno/pull/1178 - merged
- https://github.com/gnolang/gno/pull/1438 - @gfanton to make addition changes.

**Comment by thehowl** [comment]
## Session 18 -- Dec, 21st 2023

Updated PRs and issue:

- https://github.com/gnolang/gno/pull/1438 -- way forward decided
- https://github.com/gnolang/gno/pull/1177 -- reviewed with jae, looks mostly good
- https://github.com/gnolang/gno/pull/1315 -- merged

**Comment by thehowl** [comment]
## Session 19 -- Jan, 11th 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1048 -- briefly discussed with ray
- https://github.com/gnolang/gno/pull/1426 -- deferred discussion to after we have more reviews
- https://github.com/gnolang/gno/pull/1430 -- discussed, awaiting more reviews
- https://github.com/gnolang/gno/pull/1434 -- discussed, decided on a strategy

**Comment by thehowl** [comment]
## Session 20 -- Jan, 18th 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1047 -- merged
- https://github.com/gnolang/gno/pull/1200 -- blocked pending discussion w/ legal
- https://github.com/gnolang/gno/pull/1433 -- moved to awaiting changes
- https://github.com/gnolang/gno/pull/1540 -- decided not to go through, and prefer having error+fixing command workflow
- https://github.com/gnolang/gno/pull/877 -- discussed, approved the proposal
- https://github.com/gnolang/gno/pull/1553 -- merged

Also discussed:

- https://github.com/gnolang/gno/pull/1555 -- discussed how to incorporate _some_ short flags

**Comment by thehowl** [comment]
## Session 21 -- Jan, 25th 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1564 -- merged
- https://github.com/gnolang/gno/pull/1302 -- discussed, discovered aminoscan, merged
- https://github.com/gnolang/gno/pull/1574 -- merged, discussed https://github.com/gnolang/gno/pull/1583 too
- https://github.com/gnolang/gno/pull/1502 -- commented, discussed
- https://github.com/gnolang/gno/pull/1567 -- discussed

**Comment by thehowl** [comment]
## Session 22 -- Feb, 1st 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1000 -- good to merge after @leohhhn review
- https://github.com/gnolang/gno/pull/1105 -- discussed and agreed on how to proceed
- https://github.com/gnolang/gno/pull/1565 -- merged
- https://github.com/gnolang/gno/pull/1568 -- brought attention, needs reviews!
- https://github.com/gnolang/gno/pull/1434 -- pending @ajnavarro re-review
- https://github.com/gnolang/gno/pull/1608 -- merged
- https://github.com/gnolang/gno/pull/1393 -- awaiting review from @jaekwon

**Comment by thehowl** [comment]
## Session 23 -- Feb, 8th 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1613 -- merged
- https://github.com/gnolang/gno/pull/1598 -- commented, ready to merge after comments addressed
- https://github.com/gnolang/gno/pull/1430 -- discussed, still needs more reviews
- https://github.com/gnolang/gno/pull/1645 -- merge after CI
- https://github.com/gnolang/gno/pull/1624 -- merged
- https://github.com/gnolang/gno/pull/1633 -- merged
- https://github.com/gnolang/gno/pull/1606 -- awaiting @piux2 review
- https://github.com/gnolang/gno/pull/1593 -- hold off for review

**Comment by thehowl** [comment]
## Session 24 -- Feb, 15th 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1657 -- closed
- https://github.com/gnolang/gno/pull/1433 -- needs more reviews/approvals
- https://github.com/gnolang/gno/pull/1606 -- pinged @jaekwon for review
- https://github.com/gnolang/gno/pull/1048 -- concluded this discussion! finally!

Also discussed:

- https://github.com/gnolang/gno/pull/1585 (awaiting changes)
- https://github.com/gnolang/gno/pull/1593 (depends on config pr)

**Comment by thehowl** [comment]
## Session 25 -- Feb, 22nd 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1325 -- unblocked
- https://github.com/gnolang/gno/pull/1681 -- agreed on the rename
- https://github.com/gnolang/gno/pull/1598 -- agreed on renaming
- https://github.com/gnolang/gno/pull/1572 -- merge after ci
- https://github.com/gnolang/gno/pull/1568 -- given visibility
- https://github.com/gnolang/gno/pull/1678 -- merged
- https://github.com/gnolang/gno/pull/1642 -- merged
- https://github.com/gnolang/gno/pull/1606 -- merged

**Comment by thehowl** [comment]
## Session 26 -- Feb, 29th 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1601 -- visibility
- https://github.com/gnolang/gno/pull/1433 -- good to merge after brief discussion, cc/ @moul to check [this comment out](https://github.com/gnolang/gno/pull/1433/files/ff711ca5eace8fa3fc0430f55f492466f0738e68#r1507920304)
- https://github.com/gnolang/gno/pull/1602 -- discussed, asked @jaekwon for opinion
- https://github.com/gnolang/gno/pull/1672 -- visibility
- https://github.com/gnolang/gno/pull/1646 -- deferred to @jaekwon 
- https://github.com/gnolang/gno/pull/1689 -- visibility, attention to [this part](https://github.com/gnolang/gno/pull/1689/files#diff-e9e452748e81e0299fd38d1959111dd6306c9f45f8dc2ddbab2f01e3f4d01611R261-R266)

**Comment by thehowl** [comment]
## Session 27 -- Mar, 21st 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1430 -- discussed, to be tackled further
- https://github.com/gnolang/gno/pull/1807 -- discussed
- https://github.com/gnolang/gno/pull/1736 -- discussed, created an issue to point out how to move forward

**Comment by thehowl** [comment]
## Session 28 -- Mar, 28th 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1776 -- discussion of outstanding issues, and how to move forward
- https://github.com/gnolang/gno/pull/875 -- discussed what form the denom string should take
- https://github.com/gnolang/gno/pull/1495 -- discussed whether to include `ordering`
- https://github.com/gnolang/gno/pull/1714 -- discussed what implementations to keep.

**Comment by gfanton** [comment]
## Session 28 -- Apr, 04th 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1776 -- paused until amino json marshal/unmarshal implem
- https://github.com/gnolang/gno/pull/1787 -- open an issue, need second review
- https://github.com/gnolang/gno/pull/1765 -- need second review, waiting for morgan
- https://github.com/gnolang/gno/pull/1430 -- discussed on some specific jae comments

**Comment by thehowl** [comment]
## Session 29 -- Apr, 11th 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1787 -- commented and discussed
- https://github.com/gnolang/gno/pull/1777 -- agreed on alternative solution
- https://github.com/gnolang/gno/pull/1912 -- merged
- meta: let's use https://github.com/gnolang/guide/pull/39/files#diff-5d5f1505c1dea1f0d11dfeafc0840089de187b8cc879be70c0b26f23857b92f3 as a place to discuss what are the PRs required to go into the review mtg

**Comment by thehowl** [comment]
## Session 30 -- Apr, 18th 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1498 -- decided what to do re: gnofaucet
- https://github.com/gnolang/gno/pull/1944 -- decided on dir structure
- https://github.com/gnolang/gno/pull/1943 -- merge after ci
- https://github.com/gnolang/gno/pull/1702 -- decided on what to do for better client-side UX
- https://github.com/gnolang/gno/pull/1931 -- merged
- https://github.com/gnolang/gno/pull/1653 -- discussed tm2 changes

**Comment by gfanton** [comment]
## Session 31 -- Apr, 25th 2024

- https://github.com/gnolang/gno/pull/1430 -- good, need rebase
- https://github.com/gnolang/gno/pull/1939 -- good, need rebase
- https://github.com/gnolang/gno/pull/1515 -- need @jaekwon feedback
- https://github.com/gnolang/gno/pull/1919 -- assigned @petar-dambovaliev as reviewer; need some feedback from team
- https://github.com/gnolang/gno/pull/1854 -- will be reverted

**Comment by thehowl** [comment]
## Session 32 -- May, 2nd 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/2017 -- discussed map persistance
- https://github.com/gnolang/gno/issues/1886 -- discussed, awaiting manfred
- https://github.com/gnolang/gno/pull/1515 -- discussed actions (at the bottom)
- https://github.com/gnolang/gno/pull/1765 -- merged
- https://github.com/gnolang/gno/pull/1986 -- merged
- https://github.com/gnolang/gno/pull/1913#issuecomment-2091048947 -- discussion on txtar

**Comment by thehowl** [comment]
## Session 33 -- May, 9th 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/2017 -- merged
- https://github.com/gnolang/gno/pull/1965 -- merged
- https://github.com/gnolang/gno/pull/1702 -- discussed, some changes needed
- https://github.com/gnolang/gno/pull/2065 -- discussed

**Comment by thehowl** [comment]
## Session 34 -- May, 30th 2024

Updated PRs and issues:

- https://github.com/gnolang/gno/pull/1246 -- ray to take a look, merging soon
- https://github.com/gnolang/gno/pull/2129 -- discussed
- https://github.com/gnolang/gno/pull/2242 -- agreed to only have realms for `maketx call`
- https://github.com/gnolang/gno/pull/1730 -- updated

**Comment by deelawn** [comment]
## Session 35 -- June, 6th 2024
---
Updated PRs and issues:
- https://github.com/gnolang/gno/pull/1730 waiting on @thehowl 
- https://github.com/gnolang/gno/pull/2077 @petar-dambovaliev to address comments
- https://github.com/gnolang/gno/pull/2215 asked @ajnavarro what the concern is with code coverage
- https://github.com/gnolang/gno/pull/2145 @mvertes to review and comment
- https://github.com/gnolang/gno/pull/2278 merged
- https://github.com/gnolang/gno/pull/2293 merged
- https://github.com/gnolang/gno/pull/2255 needs more time for people to review
- https://github.com/gnolang/gno/pull/2281 merged
- https://github.com/gnolang/gno/pull/2156 @gfanton to add tests
- https://github.com/gnolang/gno/pull/2286 @gfanton to rebase
- https://github.com/gnolang/gno/pull/2244 need others to review
